### ISSUE TITLE
Fix large repo size

### ISSUE THREAD

**Issue by SimonBiggs** [pymedphys/pymedphys#441]
> I noticed a decent set of DICOM files are now committed. Are you sure you want to include them? I did think we were trying to minimise this sort of thing in the repo... but maybe this matters less after the big split?

It does matter, primarily for the pain it introduces for new users cloning the repo as well as every CI needs to download the latest repo version so it has pushed the cloning stage on Azure to 1 minute.

... I think we do need working examples, and I think we do need real data... I think there might be value in reviving Zenodo for file storage and then deleting every file in the git history that is > 10 MiB, and then putting commit check that won't let files > 10 MiB be commited.

I have been thinking about a neat way to ease the use of Zenodo. Essentially making a data utility package that we can use to download demo data. Have the data utility cache the downloads and have a record of the download file hash.

Even with git VFS I am beginning to suspect that might still be a neater solution.

If the tests and or docs still need those files, then they would still need to be downloaded. That's why it would be a good idea to implement a robust cache and tell Netlify as well as Azure where that cache is stored so they don't have to redownload those files every time...

_Originally posted by @SimonBiggs in https://github.com/pymedphys/pymedphys/pull/427/review_comment/create_

**Comment by SimonBiggs** [comment]
The "How-to Guides" part of the new documentation layout will inevitably become very data intensive. So need to support that well:

https://github.com/pymedphys/pymedphys/issues/416#issue-464277383

**Comment by SimonBiggs** [comment]
@Matthew-Jennings made the following comment

> If the tests and or docs still need those files, then they would still need to be downloaded. That's why it would be a good idea to implement a robust cache and tell Netlify as well as Azure where that cache is stored so they don't have to redownload those files every time...

I'm not so sure about the docs. Can the docs not point to files stored elsewhere? Then developers need not download those files to their local workstations unless they want to work on examples in particular. I don't see why developers necessarily need fully working docs locally otherwise. If they're trying to understand something, they can refer to the online docs.

**Comment by SimonBiggs** [comment]
From the documentation blog referenced within the following issue:

https://github.com/pymedphys/pymedphys/issues/416#issue-464277383

We want how-to guides in our documentation. They should be working standalone documents. My plan is to have self contained notebook files which can be downloaded from the docs page, and those notebook files should just run. Then users can see the data on their computer and swap their own data in.

**Comment by SimonBiggs** [comment]
By using a caching download solution these files will only need to download once and be stored in a PyMedPhys cache directory in the user's home directory.

**Comment by SimonBiggs** [comment]
That's where the zip files would sit, the files themselves would extract into the current working directory by default (an optionally parameter could be passed to the `get_data`  function.

**Comment by Matthew-Jennings** [comment]
I think I'm on board with this :)

**Comment by SimonBiggs** [comment]
:)

**Comment by SimonBiggs** [comment]
So, today when I sat down to do some work at home I had to go away and come back (wait quite a bit of time) just to pull in all those huge files I have committed. Git for VFS is still not cross platform, and I don't think it's worth waiting given we plan on going down the Zenodo path.

This large repo size would be stopping new comers from contributing, especially from areas with a slow internet connection.

... I propose we do something a little drastic to seriously pull back the repo size.

Firstly, implement the Zenodo solution, and then remove all files > 10 MiB from the repo. Also, any file that is not source code that is < 10 MiB might also be targeted for zendofying.

Then, mirror this repo, as is, with all of its large file history to https://github.com/pymedphys/pymedphys-archive-2019 for the ability to access the git history.

Then, clean the git history, completely, and force push a blank history to this repo with no large files in it.

Why the clean history and not use something like BFG? Because, when we did that in the past large files were still re-pushed to the repo.

Before doing this... we really want to merge all pull requests. There will be plenty of warning before doing this, and all active developers will need ample warning, and will need to reclone the repo once it is done.

I'm also open to being persuaded not to do this at all, but either way, the Zenodo solution still needs to be implemented first, so this is still a while away and discussion will be able to be undergone.
